extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java

```diff
@@ -234,4 +238,46 @@ private boolean isBinary(String contentType) {
         return false;
     }
 
+    private Optional<Principal> getPrincipal(APIGatewayV2HTTPEvent request) {
+        if (ImageInfo.inImageRuntimeCode()) {
+            return Optional
+                    .ofNullable(request.getRequestContext())
```

Comment:
While functional programming looks "pretty" to a lot of people, this code adds an extra 4 object allocations per request and consumes extra metaspace as it has to generate a class and method for each function you are invoking.

Reply:
And what is wrong with if-then-else?  sigh... :)